### ISSUE TITLE
In some cases, when multiple barracks are selected, produce option should remain disabled

### ISSUE THREAD

**Issue by andy5995** [ZetaGlest/zetaglest-source#245]
Related to #218 

If 2 or more barracks are selected, the option to produce is enabled even when there's only enough cash for one unit to be produced. The option should remain disabled, unless he has enough cash to produce from all the selected barracks.

**Comment by mathusummut** [comment]
I disagree, it should not remain disabled.

**Comment by andy5995** [comment]
The problem as I see it, is that if I have 4 barracks selected and the option is enabled, I still have to check my resources, to see how many I can produce, and then do the math in my head to know how much I need to produce 1 unit from each barracks. I don't feel it's logical or intuitive to show it enabled just because I have enough cash to produce 1. And it's not obvious to the player that it shows as enabled even if they don't have enough cash to actually build 4 if they have 4 barracks selected.

That's why I think it would be better to have it remain disabled, so a player can decide to deselect all but 1 of his barracks, or do something else and then re-select the group. (I'm usually using ctrl-grouping in this case).

So basically, I'm finding this true for me

>That would make giving commands slower and more annoying.

So if you feel the opposite, we must have different ways of producing, or you're better at doing math in your head ;)

**Comment by JoaoMarcosCSilva** [comment]
Alternatively, we can show the number of units that will be created next to (or inside) the icon.
Even though it's not as easy, we can reuse the code we make for #202.

**Comment by andy5995** [comment]
This is what that looks like in [0 A.D.](https://play0ad.com/)

![screenshot_2018-09-27_15-59-07](https://user-images.githubusercontent.com/16764864/46174350-661a7080-c26e-11e8-845c-46ad2565cb9c.png)

**Comment by mathusummut** [comment]
If we proceed as such, we would not be able to simply double-click on a building and mass-produce. We would have to unnecessarily calculate the number of buildings to select.

**Comment by JoaoMarcosCSilva** [comment]
What do you mean? It would just be a visual interface change. And even if we implement batch-training, not using it should work as usual.

**Comment by mathusummut** [comment]
I was answering @andy5995, sorry about that :p @JoaoMarcosCSilva your suggestion is best.

**Comment by JoaoMarcosCSilva** [comment]
Oh ok XD
